ui/shared/components/panel/variants/Annotations.jsx

```diff
@@ -19,28 +19,73 @@ const LargeText = styled.div`
   font-size: 1.2em;
 `
 
-const UcscBrowserLink = ({ variant, useLiftover, includeEnd }) => {
-  const chrom = useLiftover ? variant.liftedOverChrom : variant.chrom
-  const pos = parseInt(useLiftover ? variant.liftedOverPos : variant.pos, 10)
-  let genomeVersion = useLiftover ? variant.liftedOverGenomeVersion : variant.genomeVersion
-  genomeVersion = genomeVersion === GENOME_VERSION_37 ? '19' : genomeVersion
+const UcscBrowserLink = ({ genomeVersion, chrom, pos, refLength, endOffset }) => {
+  const posInt = parseInt(pos, 10)
+  const ucscGenomeVersion = genomeVersion === GENOME_VERSION_37 ? '19' : genomeVersion
 
-  const highlight = variant.ref && `hg${genomeVersion}.chr${chrom}:${pos}-${pos + (variant.ref.length - 1)}`
+  const highlight = refLength && `hg${ucscGenomeVersion}.chr${chrom}:${posInt}-${posInt + (refLength - 1)}`
   const highlightQ = highlight ? `highlight=${highlight}&` : ''
-  const endOffset = variant.end && variant.end - variant.pos
-  const position = getLocus(chrom, pos, 10, endOffset || 0)
+
+  const position = getLocus(chrom, posInt, 10, endOffset || 0)
 
   return (
-    <a href={`http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg${genomeVersion}&${highlightQ}position=${position}`} target="_blank" rel="noreferrer">
-      {`${chrom}:${pos}${(includeEnd && endOffset) ? `-${pos + endOffset}` : ''}`}
+    <a href={`http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg${ucscGenomeVersion}&${highlightQ}position=${position}`} target="_blank" rel="noreferrer">
+      {`${chrom}:${posInt}${endOffset ? `-${posInt + endOffset}` : ''}`}
     </a>
   )
 }
 
 UcscBrowserLink.propTypes = {
+  genomeVersion: PropTypes.string,
+  chrom: PropTypes.string,
+  pos: PropTypes.oneOf(PropTypes.string, PropTypes.number),
+  refLength: PropTypes.number,
+  endOffset: PropTypes.number,
+}
+
+const VariantPosition = ({ variant, svType, useLiftover }) => {
+  let { chrom, pos, end, endChrom, genomeVersion } = variant
+  const { ref, rg37LocusEnd } = variant
+
+  const showEndLocation = svType && endChrom
+  let endOffset = svType && !showEndLocation && end && end - pos
```

Comment:
yes but using a shared variable means if we ever want to tweak the logic we only need to do so in one olace. Since there is never a situation where we want to show the full end location and the end offset, making `endOffset` conditional on `showEndLocation` is much cleaner